src/Icicle/Avalanche/Eval.hs

```diff
@@ -338,9 +338,9 @@ evalStmt evalPrim now xh values bubblegum ah stmt
     Output n t xts
      -> do  vs  <- traverse ((baseValue =<<) . eval . fst) xts
             case (vs, unmeltValue vs t) of
-              (v:[], _)    -> return (ah, [(n, v)])
-              (_, Just v)  -> return (ah, [(n, v)])
-              (_, Nothing) -> Left (RuntimeErrorOutputTypeMismatch n t vs)
+              (_,    Just v)  -> return (ah, [(n, v)])
+              (v:[], Nothing) -> return (ah, [(n, v)])
```

Comment:
This is probably the most horrible part of the PR, basically this is capable of evaluating Avalanche both before and after melting. If it's been melted then unmeltValue will return Just BaseValue otherwise it will return Nothing. Nothing could mean lots of things I guess, but if we only have a single value in the list, it probably means that it was a value that didn't need unmelting because it Avalanche had not been melted yet.